pkg/ruler/ruler.go

```diff
@@ -1019,8 +1020,15 @@ func (fs StringFilterSet) IsFiltered(val string) bool {
 	return !ok
 }
 
-func (r *Ruler) getLocalRules(userID string, req RulesRequest) ([]*GroupStateDesc, error) {
+func (r *Ruler) getLocalRules(ctx context.Context, userID string, req RulesRequest) ([]*GroupStateDesc, error) {
+	spanLog, ctx := spanlogger.NewWithLogger(ctx, r.logger, "Ruler.getLocalRules")
```

Comment:
lgtm, just ctx should be discarded here.

Reply:
The reason why I'm updating it is because if in the future we'll use it, we don't forget to change this line. Basically this code is more future proof. I will make the linter stop warning, cause updating ctx is safer for the future.